jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -116,12 +134,212 @@ MlirToJlmConverter::ConvertOperation(
   else if (auto constant = ::mlir::dyn_cast<::mlir::arith::ConstantIntOp>(&mlirOperation))
   {
     auto type = constant.getType();
+    // std::cout << "Casted to ConstantIntOp with t value: " << constant.value() << "\n";
     JLM_ASSERT(type.getTypeID() == ::mlir::IntegerType::getTypeID());
-    auto integerType = ::mlir::cast<::mlir::IntegerType>(type);
+    auto integerType = type.isIntOrFloat() ? type.cast<::mlir::IntegerType>() : nullptr;
 
     return rvsdg::node_output::node(
         rvsdg::create_bitconstant(&rvsdgRegion, integerType.getWidth(), constant.value()));
   }
+
+  // Binary Comparision operations
+  else if (auto ComOp = ::mlir::dyn_cast<::mlir::arith::CmpIOp>(&mlirOperation))
```

Comment:
Maybe move the entire binary comparisons into its own function and then factor out the common code from all the cases?